### PR TITLE
Feature/mock messages count settings

### PR DESCRIPTION
**Pull request by alessioarsuffi**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Added new file to interact easily with UserDefaults;
ConversationViewController will fetch messages count set in Settings.
In SettingsViewController I've added a tableViewCell with a pickerView to select desired messages count to fetch. Starting from 0 to 99.

Does this close any currently open issues?
------------------------------------------
it is related to first point of #362 

Any relevant logs, error output, etc?
-------------------------------------

Nothing

Any other comments?
-------------------
Let me know if everything is ok for you :)

attached screen:

simulator screen shot - iphone x - 2018-01-11 at 17 41 22

Where has this been tested?
---------------------------
**Devices/Simulators:** 
iPhone 4s (iOS 9.0)
iPhone SE, 7, 7 plus, X (iOS 11)

**iOS Version:** 11.2

**Swift Version:** 4

**MessageKit Version:** 0.12.1

### CODE REVIEW

File: Example/ChatExample.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -661,10 +669,10 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				DEVELOPMENT_TEAM = 9EUCWD26TP;
+				DEVELOPMENT_TEAM = 28VWFVMBDR;
 				INFOPLIST_FILE = "$(SRCROOT)/Sources/Info.plist";
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = com.messagekit.ChatExample;
+				PRODUCT_BUNDLE_IDENTIFIER = com.messagekit.ChatExample.123;
```
Comment (SD10) [inline Example/ChatExample.xcodeproj/project.pbxproj]: Mind changing this back to `com.messagekit.ChatExample`?

Comment (alessioarsuffi) [reply Example/ChatExample.xcodeproj/project.pbxproj]: Sorry it’s a typo! Fixing ASAP

**Comment by SD10** [comment]
@MessageKit/contributors anyone want to review this one and potentially merge?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: Example/Sources/TableViewCells.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
+//
```
Comment (pryaniq) [inline Example/Sources/TableViewCells.swift]: Could you change copyright by the example of other files in the project?

File: Example/Sources/Settings+UserDefaults.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
+//
```
Comment (pryaniq) [inline Example/Sources/Settings+UserDefaults.swift]: Could you change copyright by the example of other files in the project?

File: Example/Sources/TableViewCells.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        ])
+        
+        textField.textAlignment = .right
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    // MARK: -
```
Comment (pryaniq) [inline Example/Sources/TableViewCells.swift]: Please remove it

**Review by pryaniq — CHANGES_REQUESTED** [review]
Please consider my comments. Rest of the code is normal as an example.

**Review by pryaniq — APPROVED** [review]

File: Example/Sources/SettingsViewController.swift [hunk 5]
```diff
@@ -27,8 +27,113 @@ import MessageKit
...
+    
+    var selectedMockMessagesCount: Int = 20
+    
+    // MARK: - Picker
+    
+    var messagesPicker = UIPickerView()
+    
+    @objc func onDoneWithPickerView() {
+        let selectedMessagesCount = messagesPicker.selectedRow(inComponent: 0)
+        print("selected \(selectedMessagesCount) messages count")
```
Comment (MacMeDan) [inline Example/Sources/SettingsViewController.swift]: Would you mind removing this `print` statement so the console is clean.

File: Example/Sources/SettingsViewController.swift [hunk 6]
```diff
@@ -27,8 +27,113 @@ import MessageKit
...
+    
+    // MARK: - TableViewDelegate & TableViewDataSource
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        if indexPath.row == 0 {
```
Comment (MacMeDan) [inline Example/Sources/SettingsViewController.swift]: This is not an issue just a sharing. You can exspress this entire method in one line with `shorthand` like this 
``` swift
return indexPath.row == 0 ? configureTextFieldTableViewCell(at: indexPath) : UITableViewCell()
```

File: Example/Sources/TableViewCells.swift [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ */
+
+import UIKit
+
+class TextFieldTableViewCell: UITableViewCell {
+
+    static let identifier = "TextFieldTableViewCellIdentifier"
+    static let height: CGFloat = 40.0
```
Comment (MacMeDan) [inline Example/Sources/TableViewCells.swift]: I don't believe this is used. Also it should not be necessary as of iOS 11 to specify the height of a cell. By default the tableview just uses dynamic sizing. Would you mind removing the `height`

**Review by MacMeDan — APPROVED** [review]
Just a few cleanup suggestions. Awesome work!

**Comment by alessioarsuffi** [comment]
All done @MacMeDan, thanks for you feedbacks 👍

**Comment by SD10** [comment]
@alessioarsuffi If you want you can merge this but please do a squash merge

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by alessioarsuffi** [comment]
@SD10 done, should I delete remote branch also?

**Comment by SD10** [comment]
@alessioarsuffi Yeah that's fine. It doesn't really affect MessageKit cuz its a fork. You can work on the original repo if you'd like 😃

**Comment by MacMeDan** [comment]
💯 Awesome work on this @alessioarsuffi.  💯

**Comment by alessioarsuffi** [comment]
Thanks million @MacMeDan :)

**Pull request merged**